[Docs improvement] [Update localization schema doc with v1.20 updates] (#12655)
MicrosoftDocs/msteams-docs: msteams-platform/resources/schema/localization-schema.md

description: Learn and know about the localization schema supported by the localization file for Microsoft Teams using an example schema and about schema versions.
ms.topic: reference
ms.localizationpriority: medium
ms.date: 04/14/2025
---

# Localize JSON schema reference

The Microsoft Teams localization file describes language translations that are served based on the client language settings.

> [!TIP]
> Specify the schema at the beginning of your manifest to enable `IntelliSense` or similar support from your code editor: `"$schema": "https://developer.microsoft.com/json-schemas/teams/v1.20/MicrosoftTeams.schema.json".`

## Localization schema v1.20

Your file must conform to the schema hosted at [https://developer.microsoft.com/en-us/json-schemas/teams/v1.20/MicrosoftTeams.Localization.schema.json](https://developer.microsoft.com/en-us/json-schemas/teams/v1.20/MicrosoftTeams.Localization.schema.json).

Example of localization JSON schema v1.20 is as follows:

```json
{

>     "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.19/MicrosoftTeams.Localization.schema.json",

Reviewer: @mobajemu-msft, can you confirm if the use of v1.19 was intentional or should it be updated to v1.20?

>     "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.19/MicrosoftTeams.Localization.schema.json",

Author: @v-kushals good catch, thank you.

    "staticTabs[0].name": "Accueil",
    "staticTabs[1].name": "Applications",
    "staticTabs[2].name": "Outils",
    "staticTabs[3].name": "Developer Portal",
    "bots[0].commandLists[0].commands[0].title": "Rechercher",
    "bots[0].commandLists[0].commands[0].description": "Rechercher la documentation Teams appropriée"
}
```

The schema defines the following properties:

|Property|Type|Maximum length|Description|Required|
|---------------|--------|---------|------------------|----|

> |`$schema`|URL|String|The `https://` URL referencing the JSON Schema for the manifest.||

Reviewer: ```suggestion
|`$schema`|URL|String|The `https://` URL referencing the JSON schema for the manifest.||
```

> |`name.short`|String|30|This property specifies a localized value for the [name.short property](manifest-schema.md#name). The short display name for the app. It replaces the corresponding string from the app manifest with the value provided here.|✔️|

Reviewer: ```suggestion
|`name.short`|String|30|Specifies a localized value for the [name.short property](manifest-schema.md#name). The short display name for the app. It replaces the corresponding string from the app manifest with the value provided here.|✔️|
```

Reviewer: @mobajemu-msft, this is a global suggestion for brevity and consistency with the rest of the doc set.

|`name.full`|String|100|This property specifies a localized value for the [name.full property](manifest-schema.md#name). The full name of the app. It replaces the corresponding string from the app manifest with the value provided here.|✔️|
|`description.short`|String|80|This property specifies a localized value for the [description.short property](manifest-schema.md#description). A short description of the app, used when space is limited. It replaces the corresponding string from the app manifest with the value provided here.|✔️|

> |`description.full`|String|4000|This property specifies a localized value for the [description.full property](manifest-schema.md#description). A short description of the app, used when space is limited. It replaces the corresponding string from the app manifest with the value provided here.|✔️|

Reviewer: ```suggestion
|`description.full`|String|4000|This property specifies a localized value for the [description.full property](manifest-schema.md#description). A full description of the app, used when space isn't limited. It replaces the corresponding string from the app manifest with the value provided here.|✔️|
```

|`localizationKeys`|Object||Represents custom tokenized keys for localized strings in agents. Each key is represented by a property name that matches a regular expression (with the following format: `^\[\[[a-zA-Z_][a-zA-Z0-9_]*\]\]$`) and the value provides the localized string value. For more information, see [localize your agent](/microsoft-365-copilot/extensibility/agents-are-apps#localizing-your-agent).||

> |`staticTabs\\[([0-9]/1[0-5])\\]\\.name`|String|128|This property specifies a localized value for the [staticTabs.name property](manifest-schema.md#statictabs). The property name should be a JSON path expression in the following form: staticTabs[0-15].name.||

Reviewer: ```suggestion
|`staticTabs\\[([0-9]/1[0-5])\\]\\.name`|String|128|This property specifies a localized value for the [staticTabs.name property](manifest-schema.md#statictabs). The property name should be a JSON path expression in the following form: `staticTabs[0-15].name.`||
```

Reviewer: @mobajemu-msft, can we add code tags for all the JSON path expression formats? Just splitting hairs here, but should it be "form" or "format"? Please make the change as you see fit.
